src/main/java/com/learning/mfscreener/service/PortfolioService.java

```diff
@@ -40,45 +36,39 @@ public class PortfolioService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(PortfolioService.class);
 
-    private final ObjectMapper objectMapper;
     private final ConversionServiceAdapter conversionServiceAdapter;
     private final CasDetailsMapper casDetailsMapper;
-    private final UserCASDetailsEntityRepository casDetailsEntityRepository;
     private final InvestorInfoEntityRepository investorInfoEntityRepository;
-    private final NavService navService;
     private final SchemeService schemeService;
-    private final XIRRCalculatorService xIRRCalculatorService;
     private final UserTransactionDetailsService userTransactionDetailsService;
     private final UserFolioDetailsService userFolioDetailsService;
     private final UserSchemeDetailsService userSchemeDetailsService;
+    private final PortfolioServiceHelper portfolioServiceHelper;
+    private final UserCASDetailsService userCASDetailsService;
 
     public PortfolioService(
-            ObjectMapper objectMapper,
             ConversionServiceAdapter conversionServiceAdapter,
             CasDetailsMapper casDetailsMapper,
-            UserCASDetailsEntityRepository casDetailsEntityRepository,
             InvestorInfoEntityRepository investorInfoEntityRepository,
             UserTransactionDetailsService userTransactionDetailsService,
-            NavService navService,
             SchemeService schemeService,
-            XIRRCalculatorService xIRRCalculatorService,
             UserFolioDetailsService userFolioDetailsService,
-            UserSchemeDetailsService userSchemeDetailsService) {
-        this.objectMapper = objectMapper;
+            UserSchemeDetailsService userSchemeDetailsService,
+            PortfolioServiceHelper portfolioServiceHelper,
+            UserCASDetailsService userCASDetailsService) {
         this.conversionServiceAdapter = conversionServiceAdapter;
         this.casDetailsMapper = casDetailsMapper;
-        this.casDetailsEntityRepository = casDetailsEntityRepository;
+        this.userCASDetailsService = userCASDetailsService;
         this.investorInfoEntityRepository = investorInfoEntityRepository;
-        this.navService = navService;
         this.schemeService = schemeService;
-        this.xIRRCalculatorService = xIRRCalculatorService;
         this.userFolioDetailsService = userFolioDetailsService;
         this.userSchemeDetailsService = userSchemeDetailsService;
         this.userTransactionDetailsService = userTransactionDetailsService;
+        this.portfolioServiceHelper = portfolioServiceHelper;
     }
 
     public String upload(MultipartFile multipartFile) throws IOException {
```

Comment:
@coderabbitai, suggest code changes without breaking functionality to reduce cognitive complexity.